lib/arel/collectors/sunstone.rb

```diff
@@ -75,8 +75,19 @@ def compile bvs, conn = nil
           get_params[:include] = eager_loads.clone
         end
 
-        get_params[:limit] = substitute_binds(limit, bvs) if limit
-        get_params[:offset] = substitute_binds(offset, bvs) if offset
+
+        if limit.is_a?(Arel::Nodes::BindParam)
+          get_params[:limit] = substitute_binds(limit, bvs)
+        elsif limit
+          get_params[:limit] = limit
+        end
```

Comment:
This would just be replacing the BindParam, so I guess we could replace the value in `bvs`? This way is fine I think though.